csrc/id_model/indexing_traversal.cpp

```diff
@@ -0,0 +1,46 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <id_model/indexing_traversal.h>
+
+namespace nvfuser {
+
+IndexingTraversal::IndexingTraversal(
+    const Expr* expr,
+    const ValGraph& graph,
+    std::vector<NodeType> from_groups,
+    std::vector<NodeType> to_groups)
+    : ValGraphBFS(graph, from_groups, to_groups) {
+  auto consumer_tv = ir_utils::getTvOutput(expr);
+  NVF_ERROR(consumer_tv != nullptr);
+  // Remember the resize exprs appearing in the consumer
+  // tensor. These resize exprs are the only ones that should be
+  // valid to visit when indexing the inputs and outputs of the
+  // expr.
+  //
+  // This is a WAR for cases like
+  // ResizeTest.SliceScheduledLikeProducer. Alternatively, we could
+  // have a separate graph for indexing that does not map producer and
+  // consumers in non-unary ops. See PR #2897.
+  auto all_ids = consumer_tv->domain()->allIDs();
+  std::unordered_set<IterDomain*> all_id_set(all_ids.begin(), all_ids.end());
+  for (auto id : all_ids) {
+    auto resize = dynamic_cast<Resize*>(id->definition());
+    if (resize == nullptr) {
+      continue;
+    }
+    auto resize_in = resize->in();
+    if (all_id_set.find(resize_in) == all_id_set.end()) {
+      // ths resize must not be part of the exprs involved for
+      // the domains of consumer_tv
+      continue;
+    }
```

Comment:
It's good to be safe here, but I wonder how this condition could fire. When would we have a `Resize` whose output is in the `TensorDomain` but its input is not?

Reply:
It's unlikely, but for example, I'm replacing the definition of an ID with a resize:

https://github.com/NVIDIA/Fuser/pull/2898/files#diff-98e2e8f5cb3849584331987b87d7cc114cb88cfddddd2f739ca8b452d92f4eceR3866-R3870

In this case the resize is in the path between the root and loop domains. Technically, it's a valid operation to change the loop domain again, and suppose its loop domain is changed back to its logical. The root domain would still keep the resize as its definition, but it is no longer part of the valid operations of the tensor, so it would need to be filtered out. Alternatively, we could delete the resize definition when the loop domain is changed.